File: src/Moq/Moq.Package/Mock.Overloads.vb
Imports System.Runtime.CompilerServices

Namespace Global.Moq

    Partial Friend Class Mock

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs)
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1), GetType(T2))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1), GetType(T2), GetType(T3))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4, T5)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4), GetType(T5))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4, T5, T6)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4), GetType(T5), GetType(T6))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4, T5, T6, T7)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4), GetType(T5), GetType(T6), GetType(T7))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4, T5, T6, T7, T8)(ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4), GetType(T5), GetType(T6), GetType(T7), GetType(T8))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(behavior, constructorArgs)
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(behavior, constructorArgs, GetType(T1))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(behavior, constructorArgs, GetType(T1), GetType(T2))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(behavior, constructorArgs, GetType(T1), GetType(T2), GetType(T3))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(behavior, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4, T5)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(MockBehavior.Loose, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4), GetType(T5))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4, T5, T6)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(behavior, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4), GetType(T5), GetType(T6))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4, T5, T6, T7)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(behavior, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4), GetType(T5), GetType(T6), GetType(T7))
        End Function

        <MockGenerator>
        <CompilerGenerated>
        Public Shared Function [Of](Of T As Class, T1, T2, T3, T4, T5, T6, T7, T8)(ByVal behavior As MockBehavior, ParamArray constructorArgs As Object()) As T
            Return Create(Of T)(behavior, constructorArgs, GetType(T1), GetType(T2), GetType(T3), GetType(T4), GetType(T5), GetType(T6), GetType(T7), GetType(T8))
        End Function

    End Class

End Namespace

File: src/Moq.Package/Mock.Overloads.cs
using System.Runtime.CompilerServices;

namespace Moq
{
    /// <summary>
    /// Instantiates stunts for the specified types.
    /// </summary>
    partial class Mock
    {
        /// <summary>
        /// Creates a mock that inherits or implements the type <typeparamref name="T"/>.
        /// </summary>
        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs);

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs, typeof(T1));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs, typeof(T1), typeof(T2));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs, typeof(T1), typeof(T2), typeof(T3));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4, T5>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4, T5, T6>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5), typeof(T6));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4, T5, T6, T7>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5), typeof(T6), typeof(T7));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4, T5, T6, T7, T8>(params object[] constructorArgs) where T : class => Create<T>(MockBehavior.Loose, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5), typeof(T6), typeof(T7), typeof(T8));

        /// <summary>
        /// Creates a mock that inherits or implements the type <typeparamref name="T"/>.
        /// </summary>
        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs);

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs, typeof(T1));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs, typeof(T1), typeof(T2));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs, typeof(T1), typeof(T2), typeof(T3));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4, T5>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4, T5, T6>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5), typeof(T6));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4, T5, T6, T7>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5), typeof(T6), typeof(T7));

        [MockGenerator]
        [CompilerGenerated]
        public static T Of<T, T1, T2, T3, T4, T5, T6, T7, T8>(MockBehavior behavior, params object[] constructorArgs) where T : class => Create<T>(behavior, constructorArgs, typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5), typeof(T6), typeof(T7), typeof(T8));
    }
}